_Event: \{\{eventName\}\}_

# Reproducibility for Everyone

# {{firstInstructorName}}
{{secondInstructorName}}
{{thirdInstructorName}}
{{fourthInstructorName}}

# Slide deck: {{slidesLink}}
Shared notes: {{sharedNotesLink}}
Handout: {{handoutLink}}

\[firstInstructorPhoto\]

# Hello!

# I am 
{{firstInstructorName}}

\{\{firstInstructor __Affiliation__ \}\}

\{\{firstInstructorContact\}\}

# Who are we?

* __Community of scientists in academia and industry who want to improve the reproducibility of science__
  * __Members all over the world\!__
  * __Members help in developing new slides\,__  __hosting workshops\, applying for funding\, etc\.__
* __Developed this__  __open access__  __workshop\, which you can use yourself for a journal club / seminar at your institute\!__
* __Want to join us?__  __Reach out to the workshop organizers or through our website:__  _[https://www\.repro4everyone\.org/join](https://www.repro4everyone.org/join)_

_[https://repro4everyone\.org/](https://repro4everyone.org/)_

# Thank You

![](img/main-slides_full0.png)

![](img/main-slides_full1.jpg)

![](img/main-slides_full2.png)

![](img/main-slides_full3.png)

Benjamin Schwessinger

@schwessinger

![](img/main-slides_full4.png)

_@repro4everyone_

_hello@repro4everyone\.org_

_www\.repro4everyone\.or_  _g_

# If you want to read more about our initiative:

![](img/main-slides_full5.png)

_[https://osf\.io/dxw67/](https://osf.io/dxw67/)_

__Participants agree to follow the R4E__  __Community Participation Guidelines__  _[repro4everyone\.org/pages/guidelines](https://repro4everyone.org/pages/guidelines/)_

__DO__

__✅__ Be respectful

__✅	Give everyone a chance to contribute__

__✅__ Use inclusive language

__✅__ Appreciate and accommodate differences

__✅__ Lead by example

__DON'T__

__❌__ Repeatedly interrupt or disrupt others

__❌	Use s__ exual language or imagery

__❌	Give__ unwelcome attention

__❌__ Bully\, discriminate\, or harass

__❌	Make fun of personal appearance or choices__

__Have a concern?__

Report by sending a private Zoom chat message to Nele\.

Report by emailing _[coc@repro4everyone\.org](mailto:coc@repro4everyone.org)_ or anonymously at _[https://forms\.gle/UaxjwEYWVNoCDwJs5](https://forms.gle/UaxjwEYWVNoCDwJs5)_

__Violations may result in muting or removing of a participant\.__

# Communication during webinar

* __Questions__ :
  * __Type your question__ into the related section of the shared notes document\.
  * __\+1 questions__ that you like or think are important\.
  * Know the answer to a question? Type it below the question\.

![](img/main-slides_full6.png)

* __Comments__ :
  * Have __experiences or tips__ related to the topic? Type them into the related section of the shared notes document\.
  * __\+1 comments__ that you think were useful\.

# Introduction

# Why does reproducibility matter to you?

# Have you ever had problems reproducing your own or someone else’s research?

![](img/main-slides_full7.png)

# Goals and Objectives

‘Reproducibility’ framework

‘Reproducibility’ tools

Starting point for a ‘lifelong’ journey

# We will look at...

What does reproducibility mean?

What are the different modes of reproducibility?

Is reproducibility all that matters?

‘Reproducibility tool shed:

  * __Organization__
  * __Documentation__

  * __Analysis__
  * __Dissemination__

# What does reproducibility mean?

_Reproducible research_ : Authors provide all the necessary data and the computer codes to run the analysis again\, re\-creating the results\.

_Replication_ : A study that arrives at the same scientific findings as another study\, collecting new data and completing new analyses\.

__Barba\, 2018 \(__  _[https://arxiv\.org/abs/1802\.03311](https://arxiv.org/abs/1802.03311)_  __\)__

__Schloss\, 2018 \(__  _[10\.1128/mBio\.00525\-18](https://mbio.asm.org/content/mbio/9/3/e00525-18.full.pdf)_  __\)__

# Different modes of ‘reproducibility’

__The Turing Way \(__  _[https://the\-turing\-way\.netlify\.app/](https://the-turing-way.netlify.app/)_  __\)__

__Schloss\, 2018 \(__  _[10\.1128/mBio\.00525\-18](http://mbio.asm.org/content/9/3/e00525-18.abstract)_  __\)__

# Is reproducibility all that matters?

![](img/main-slides_full8.png)

__Casadevall and Fang\, 2016 \(__  _[10\.1128/mBio\.01902\-16](https://mbio.asm.org/content/9/1/e00043-18.full.pdf)_  __\)__

![](img/main-slides_full9.png)

_Every little bit helps\!_

_No one is perfect\!_

_Transparent and open science\!_

_Everyone starts somewhere\!_

__Casadevall and Fang\, 2016 \(__  _[10\.1128/mBio\.01902\-16](https://mbio.asm.org/content/9/1/e00043-18.full.pdf)_  __\)__

# Factors decreasing reproducibility

![](img/main-slides_full10.png)

![](img/main-slides_full11.png)

![](img/main-slides_full12.png)

![](img/main-slides_full13.png)

![](img/main-slides_full14.png)

![](img/main-slides_full15.png)

![](img/main-slides_full16.png)

# What can we do by 
the end of the century?

Cori Bargmann\, Howard Hughes Medical Institute Investigator & Chan Zuckerberg Initiative President of Science:

“82 years ago\, there were no antibiotics and we didn’t know that smoking causes lung cancer…

We can expect a lot from the next 82 years\.”

<span style="color:#FFFFFF"> __Where can we be in 82 years__ </span>

<span style="color:#FFFFFF"> __if we__ </span>  <span style="color:#FFFFFF"> _accelerate_ </span>  <span style="color:#FFFFFF"> __science?__ </span>

Cori Bargmann\, HHMI Investigator & President CZI Science:

“82 years ago\, there were no antibiotics and we didn’t know that smoking causes lung cancer…

We can expect a lot from the next 82 years\.”

# Where is your greatest potential for growth?

# What are major improvements that could lead to more reproducibility?

![](img/main-slides_full17.png)

Better \+ more detailed methods

Open data \+ metadata

Increased statistical power

Less pressure to publish

Fewer incentives to be first

Better reagent sharing

Better code sharing

# Everyone starts somewhere!

# Every little bit helps!

# No one is perfect!

# Everyone starts somewhere!

# Every little bit helps!

# No one is perfect!

<span style="color:#FFFFFF"> __Adopting some of these best practices__ </span>

<span style="color:#FFFFFF"> __isn’t just good for other scientists…__ </span>

<span style="color:#FFFFFF"> __It’s good for you &__ </span>

<span style="color:#FFFFFF"> __will save you time in the long term\!__ </span>

\{\{secondInstructorPhoto\}\}

# Hello!

# I am 
{{secondInstructorName}}

\{\{secondInstructorAffiliation\}\}

\{\{secondInstructorContact\}\}

# Data Management

# Are you familiar with these feelings?

_I cannot find this file\!_

_What did I call it again??_

![](img/main-slides_full18.png)

_Was it the wild type picture or the mutant one?_

_Where is my file??_

_What version was it??_

_Where is my RAW\!\!\! data???_

# Make a plan! 
Be happy!

<span style="color:#CFE2F3">No need to become a science superhero overnight\!</span>

![](img/main-slides_full19.png)

![](img/main-slides_full20.png)

# Think about...

![](img/main-slides_full21.png)

<span style="color:#333333"> __What__ </span>  <span style="color:#333333">data will be produced as a part of the project?</span>

<span style="color:#333333"> __How__ </span>  <span style="color:#333333">each type of data will be organized\, documented\, standardized\, stored\, protected\, shared and archived?</span>

<span style="color:#333333"> __Who__ </span>  <span style="color:#333333">will take responsibility for carrying out the activities listed above\, and</span>

<span style="color:#333333"> __When__ </span>  <span style="color:#333333">these activities will take place over the course of the project \(and beyond\)?</span>

<span style="color:#333333"> __Metadata__ </span>

_[guides\.lib\.purdue\.edu/c\.php?g=353013&p=2378292](http://guides.lib.purdue.edu/c.php?g=353013&p=2378292)_ \, _[www\.dataone\.org/best\-practices](https://www.dataone.org/best-practices)_

<span style="color:#000000"> __Project directory structure__ </span>

# Develop an informative directory structure
Keep research materials together

* <span style="color:#000000">Project\_1</span>
* <span style="color:#000000">methods</span>
* <span style="color:#000000">raw\_data</span>
* <span style="color:#000000">analysis</span>
* <span style="color:#000000">scripts</span>
* <span style="color:#000000">manuscript</span>
* <span style="color:#000000">readme and/or ELN link</span>

<span style="color:#000000">Inspired by</span>  _[‘Bioinformatic data skills’](http://shop.oreilly.com/product/0636920030157.do)_

<span style="color:#000000">by Vincent Buffalo</span>

<span style="color:#000000"> __Project directory structure__ </span>

_Specific content in each category for Project \#1_

* <span style="color:#000000">Project\_1</span>
* <span style="color:#000000">methods</span>
* <span style="color:#000000">raw\_data</span>
  * <span style="color:#000000">readme</span>
* <span style="color:#000000">analysis</span>
  * <span style="color:#000000">analysis\_method\_1</span>
    * <span style="color:#000000">2017</span>
    * <span style="color:#000000">2018</span>
  * <span style="color:#000000">analysis\_method\_2</span>
* <span style="color:#000000">scripts</span>
* <span style="color:#000000">manuscript</span>
  * <span style="color:#000000">text</span>
    * <span style="color:#000000">version\_1</span>
* <span style="color:#000000">readme and/or ELN link</span>

__Raw data\,__

__Data analysis\, and Manuscript__

<span style="color:#000000">Inspired by</span>  _[‘Bioinformatic data skills’](http://shop.oreilly.com/product/0636920030157.do)_

<span style="color:#000000">by Vincent Buffalo</span>

# Project directory structure

* __Project\_1__
  * __methods__
  * __raw\_data__
    * readme
  * __analysis__
    * analysis\_method\_1
      * 2017
      * 2018
    * Analysis\_method\_2
  * __scripts__
  * __manuscript__
    * text
      * Version\_1
  * __readme and/or ELN link__

<span style="color:#FFFFFF">Always keep</span>  <span style="color:#FFFFFF"> __raw__ </span>  <span style="color:#FFFFFF">data\!</span>

<span style="color:#FFFFFF">Always</span>  <span style="color:#FFFFFF">b</span>  <span style="color:#FFFFFF">ackup</span>  <span style="color:#FFFFFF">d</span>  <span style="color:#FFFFFF">ata</span>

<span style="color:#FFFFFF">\(3x and synchronized: 3 unique locations \- cloud\, server\,</span>

<span style="color:#FFFFFF">personal drive\)</span>

<span style="color:#000000">Inspired by</span>  _[‘Bioinformatic data skills’](http://shop.oreilly.com/product/0636920030157.do)_

<span style="color:#000000">by Vincent Buffalo</span>

# File naming conventions

![](img/main-slides_full22.png)

What did you call the last file you generated?

Did you have rules?

_[http://guides\.lib\.purdue\.edu/c\.php?g=353013&p=2378292](http://guides.lib.purdue.edu/c.php?g=353013&p=2378292)_

_[http://kbroman\.org/dataorg/](http://kbroman.org/dataorg/)_

_The rules don’t matter; that you have rules matters_

Include date in yyyy\-mm\-dd format

Use meaningful abbreviations

Have group identifiers

Document your decisions

Be consistent

Use version numbers

_[guides\.lib\.purdue\.edu/c\.php?g=353013&p=2378292](http://guides.lib.purdue.edu/c.php?g=353013&p=2378292)_ \, _[kbroman\.org/dataorg/](http://kbroman.org/dataorg/)_

Example

_20130825_  <span style="color:#333333">\_</span>  <span style="color:#F20253">DOEProject</span>  <span style="color:#333333">\_</span>  _Ex1Test1_  <span style="color:#333333">\_</span>  <span style="color:#F20253">Data</span>  <span style="color:#333333">\_</span>  _Gonzalez_  <span style="color:#333333">\_</span>  <span style="color:#F20253">v3\-03</span>  <span style="color:#333333">\.xlsx</span>

_[http://guides\.lib\.purdue\.edu/c\.php?g=353013&p=2378292](http://guides.lib.purdue.edu/c.php?g=353013&p=2378292)_

# Electronic Notebooks

# Paper Lab-notebooks - in use since the 15th Century!

![](img/main-slides_full23.png)

Good record keeping is important for:

Dissemination of ideas\, findings\,

Legally binding records that protects intellectual property

<span style="color:#000000">Not searchable\!</span>

<span style="color:#000000">Can be easily damaged\, misplaced</span>

<span style="color:#000000">Not easy to back up</span>

<span style="color:#000000">Hard to share with collaborators</span>

__Leonardo da Vinci’s notebook\,__

__Codex Arundel c\. 1458\-1518__

__British Library__

# Why should you use an Electronic Lab Notebook?

__Search__

Easily searchable both in online and offline searches

__Export__

Data can be exported as PDF \(must backup regularly\)

__Share__

Easily shareable with collaborators and broader community

__Embed__

Can embed high res images\, protocols\, & more

__Mobile__

Use the mobile app to quickly upload images

__Access__

Easily accessible globally \(depend on connectivity\)

# Basic features of an Electronic Lab Notebook

![](img/main-slides_full24.png)

# Cost considerations - Available Products

Paid for \- Bio\-Itech\, LabArchives\, LabGuru

Paid \(with free version\) \- SciNote\, Benchling

Open source \- Open wet ware\, ELOG

Free \- Open Science Framework \(OSF\.io\)\, LocalWiki

<span style="color:#000000">Kanza\,</span>  _[10\.1186/s13321\-017\-0221\-3](https://jcheminf.biomedcentral.com/articles/10.1186/s13321-017-0221-3)_

# One size does not fit all

<span style="color:#FFFFFF"> __Available lab notebooks__ </span>

![](img/main-slides_full25.png)

<span style="color:#FFFFFF"> __Parameters__ </span>

<span style="color:#FFFFFF"> __to consider__ </span>

<span style="color:#FFFFFF"> __ELN Features__ </span>

<span style="color:#FFFFFF"> __Matrix__ </span>

_[https://datamanagement\.hms\.harvard\.edu/electronic\-lab\-notebooks](https://datamanagement.hms.harvard.edu/electronic-lab-notebooks)_

# General tips on electronic record keeping

Back\-up data regularly

Maintain a physical observation notebook in parallel

Mobile apps provide added portability

If using free ELNs\, check privacy and data ownership policies

# Organizing and Sharing Protocols

# Being able to find optimized protocols for your organism / tissue of choicecan save you months!

![](img/main-slides_full26.jpg)

_[https://medium\.com/@tpoi](https://medium.com/@tpoi)_

I create them myself

# Where do you find your protocols?

Contacting an expert in the field

![](img/main-slides_full27.png)

# Description 
Unavailable

![](img/main-slides_full28.png)

![](img/main-slides_full29.png)

![](img/main-slides_full30.gif)

![](img/main-slides_full31.png)

![](img/main-slides_full32.png)

# Description
Ambiguous

![](img/main-slides_full33.png)

![](img/main-slides_full34.png)

_[https://medium\.com/@tpoi](https://medium.com/@tpoi)_

# Use repositories, not supp. files

![](img/main-slides_full35.png)

![](img/main-slides_full36.png)

<span style="color:#0097A7"> _[https://protocolexchange\.researchsquare\.com](https://protocolexchange.researchsquare.com/)_ </span>

<span style="color:#0097A7"> _[https://www\.protocols\.io](https://www.protocols.io/)_ </span>

![](img/main-slides_full37.jpg)

<span style="color:#0097A7"> _[https://theplosblog\.plos\.org/2020/12/show\-your\-work\-peer\-reviewed\-protocols/](https://theplosblog.plos.org/2020/12/show-your-work-peer-reviewed-protocols/)_ </span>

![](img/main-slides_full38.png)

![](img/main-slides_full39.png)

![](img/main-slides_full40.png)

![](img/main-slides_full41.png)

![](img/main-slides_full42.png)

![](img/main-slides_full43.png)

![](img/main-slides_full44.png)

<span style="color:#000000"> __Accelerate Science__ </span>

→ Increase Discoverability <span style="color:#000000">→ Reproducibility</span>

→ Facilitate Research Connections

→ Enable Reuse

<span style="color:#000000">→ Enhance Value of Research</span>

![](img/main-slides_full45.png)

![](img/main-slides_full46.png)

![](img/main-slides_full47.png)

# Protocol drawing exercise

![](img/main-slides_full48.png)

![](img/main-slides_full49.png)

__1\.__ Draw a small 8/10th of an oval in the center of your paper with the open part facing to the bottom left\. __2\.__ Draw ⅓ of a large circle underneath the oval and draw small perpendicular lines on each end of it\. __3\.__ At the center of the ⅓ circle\, draw two squares with a small gap between them\. __4\.__ Draw a wavy line underneath the squares\. __5\.__ On the ends of the larger circle\, draw smaller half circles around them with three dots in the middle\. __6\.__ Draw two big circles on each side of the small oval from the beginning\. They should be more on the top than on the bottom\. __7\.__ In the big circles\, draw smaller circles and in the smaller circles draw even smaller filled in circles\. __8\.__ Draw a wavy square around everything you drew so far\. __9\.__ Draw 7 small ovals within the space of the square\.

![](img/main-slides_full50.jpg)

![](img/main-slides_full51.png)

![](img/main-slides_full52.jpg)

![](img/main-slides_full53.jpg)

<span style="color:#000000">One protocol</span>

<span style="color:#000000">\- many results\!</span>

![](img/main-slides_full54.jpg)

![](img/main-slides_full55.jpg)

![](img/main-slides_full56.jpg)

![](img/main-slides_full57.jpg)

![](img/main-slides_full58.jpg)

![](img/main-slides_full59.jpg)

![](img/main-slides_full60.png)

![](img/main-slides_full61.jpg)

![](img/main-slides_full62.jpg)

![](img/main-slides_full63.jpg)

# Detail & Digitize Protocols

* Think of protocol as brief\, modular\, _self\-contained_ scientific publication
* Include 3\-4 sentence abstract that _puts methodology in context_
* Include _as much detail as possible_
  * duration/time per step
  * reagent amount
  * vendor name
  * catalogue number

__Expected result__

__Safety information__

__Software package__

Chronology of steps

Notes\, recipes\, tips and tricks

Use protocols\.io\, Google Docs\, and/or ELN \(not paper or MS Word\) \- __need versioning__

_[https://www\.protocols\.io/view/how\-to\-make\-your\-protocol\-more\-reproducible\-discov\-g7vbzn6](https://www.protocols.io/view/how-to-make-your-protocol-more-reproducible-discov-g7vbzn6)_  _[https://www\.aje\.com/en/arc/how\-to\-write\-an\-easily\-reproducible\-protocol/](https://www.aje.com/en/arc/how-to-write-an-easily-reproducible-protocol/)_

\{\{thirdInstructorPhoto\}\}

# Hello!

# I am 
{{thirdInstructorName}}

\{\{thirdInstructor __Affiliation__ \}\}

\{\{thirdInstructorContact\}\}

# ‘Wet lab’ Reagent Sharing

# Problems with wet-lab reagent availability

![](img/main-slides_full64.png)

![](img/main-slides_full65.png)

![](img/main-slides_full66.png)

![](img/main-slides_full67.png)

<span style="color:#000000">Scientist makes and publishes a reagent</span>

<span style="color:#000000"> __“available upon request”__ </span>

<span style="color:#000000">Scientist leaves the lab and stores published reagent in freezer</span>

<span style="color:#000000">Other scientists request the</span> reagent <span style="color:#000000">\, but no one else remembers where it is stored\!</span>

![](img/main-slides_full68.png)

In 2005\, NIH Study found that <span style="color:#000000">half of the mouse lines had been remade at least twice</span>

<span style="color:#000000">Only 12% were available from repositories</span>

It can take years and cost $$$ for researchers to make a mouse strain\.

<span style="color:#000000">Lloyd et al\. Nature 2015</span>  <span style="color:#000000"> _[10\.1038/522151a](https://dx.doi.org/10.1038%2F522151a)_ </span>

![](img/main-slides_full69.png)

<span style="color:#666666">doi:10\.1126/science\.aau9619</span>

* __Requesting researcher:__
  * __Waiting to receive the reagent from corresponding author delays research__
  * __Having to recreate it themselves can introduce errors which can lead to inaccurate or misleading results\.__
* __Corresponding author:__  __Individual labs don’t always have the resources to__
  * __Properly track and store all reagents__
  * __Consistently authenticate all reagents in the lab__
  * __\(Legally\) distribute all reagents to requesting researchers__
  * __Ability to navigate shipping and export/import regulations__
* __Reagents repositories are part of the solution\!__

# Functions of reagent repositories

Speedier access to materials

Authenticate reagents and maintain quality control

Curate reagents and standardize reagent information

Facilitate and track shipping

![](img/main-slides_full70.png)

![](img/main-slides_full71.png)

![](img/main-slides_full72.png)

# How to improve reagent sharing & reporting

<span style="color:#000000">Record how a reagent was created</span>

Authenticate\!

__Provide associated publications & protocols__

<span style="color:#000000">When naming reagents you make\, use descriptive and standardized naming conventions</span>

<span style="color:#000000">Record and include the reagent’s catalog \# or RRID in your</span> protocols\, <span style="color:#000000">lab notebooks and manuscript\(s\)</span>

<span style="color:#000000">Deposit reagents & organisms you</span> make <span style="color:#000000">with a repository\!</span>

![](img/main-slides_full73.png)

# Benefits of depositing reagents

__DIRECT__

Archiving

Reducing time spent sending out reagents

Authentication/Quality Control

__INDIRECT__

Creation of educational content

Direct promotion

Analysis of reagent distribution

![](img/main-slides_full74.png)

_[https://blog\.addgene\.org/addgene\-depositors\-get\-more\-citations](https://blog.addgene.org/addgene-depositors-get-more-citations)_

# Examples of reagent/organism repositories

![](img/main-slides_full75.png)

Addgene

ATCC \(American Type Culture Collection\)

ABRC \(Arabidopsis Biological Resource Center\)

Bloomington Drosophila Stock Center

Caenorhabditis Genetics Center

Developmental Studies Hybridoma Bank

EUROSCARF \(EUROpean Saccharomyces Cerevisiae ARchive for Functional analysis\)

MMRRC \(Mutant Mouse Resource and Research Centers\)

National Xenopus Resource \(NXR\)

ZIRC \(Zebrafish International Resource Center\)

![](img/main-slides_full76.png)

![](img/main-slides_full77.png)

![](img/main-slides_full78.png)

![](img/main-slides_full79.png)

# Addgene: The nonprofit biomaterials repository

__Goal:__ To accelerate science by improving access to research materials and information

![](img/main-slides_full80.png)

__Issues Addressed:__ Difficulties in obtaining\, verifying\, and using plasmids from other labs

__Audience:__ Academic\, nonprofit\, and industry scientists doing biology research and using plasmids and viral vectors\.

__Services:__

Stores and distributes plasmids and prepared viral vectors

Verifies plasmids and viral vectors through DNA sequencing with some functional testing

Collates/curates information about plasmids and viral vectors

Produces and freely distributes educational content to make it easier for scientists to learn about and use new technologies

# Summary of reagent sharing

Sharing increases access and speeds up science\.

Record key details about reagents you are using\!

Deposit reagents you make\! Improves reagent sharing and reproducibility\.

\{\{fourthInstructorPhoto\}\}

# Hello!

# I am 
{{fourthInstructorName}}

\{\{fourthInstructor __Affiliation__ \}\}

\{\{fourthInstructorContact\}\}

# Bioinformatics Tools
for computational reproducibility

# What version of the program, data, etc… did I use? Why won’t it work?

__Why did I do this?__

<span style="color:#000000">Dependency</span> hell <span style="color:#000000">\(files require other files\)</span>

![](img/main-slides_full81.png)

![](img/main-slides_full82.png)

_[https://carpentries\.org](https://carpentries.org)_  _[‘Bioinformatic data skills’](http://shop.oreilly.com/product/0636920030157.do)_  __Vincent Buffalo__

# Document your analysis with literate programming

Documentation of your analysis narrative and the analysis code together in one executable document

What you did and why you did it

Interactive data exploration

Easily shared

Best start: Jupyter Notebooks or RMarkdown with KnitR

![](img/main-slides_full83.png)

_[https://jupyter\.org/](https://jupyter.org/)_

![](img/main-slides_full84.png)

_[https://www\.rstudio\.com](https://www.rstudio.com)_

# Jupyter Notebooks - Example

![](img/main-slides_full85.png)

# Document changes

![](img/main-slides_full86.png)

<span style="color:#FFFFFF">With version control</span>

<span style="color:#FFFFFF">Git</span>

Records changes \(what\, when\, who\)

Documents version history

Illustrates changes between versions \(diffs\)

<span style="color:#FFFFFF">Github</span>

<span style="color:#FFFFFF">Lets you share code easily</span>

<span style="color:#FFFFFF">Lets you collaborate on your code more easily</span>

![](img/main-slides_full87.png)

__Version\-control__

__system that__

__uses__  _forks_  __and__  _pull requests_  __\.__

<span style="color:#000000">Git\-based repository with</span>

<span style="color:#000000">a social media component\.</span>

_[http://smutch\.github\.io/VersionControlTutorial//](http://smutch.github.io/VersionControlTutorial//)_

# How did I install all these different software packages???

Version conflict

![](img/main-slides_full88.png)

![](img/main-slides_full89.png)

# Document your computational environment with package, dependency & environment managers

![](img/main-slides_full90.png)

Python virtual environments

Dependency managers

Handles installs & dependencies

Document your environment \(requirements\.txt\)

__In RStudio\, use CRAN and RProjects__

__Document your packages \(install\.R\)__

_[https://conda\.io/](https://conda.io/)_  _[https://bioconda\.github\.io/](https://bioconda.github.io/)_

# Make your environment portable with containers

Everything needed to run your analysis is packed up into an “image”

Images are self\-contained with all code\, programs\, environment\, Dockerfile included

No subsequent installation required

Spin an image into a container using Docker or Singularity \- it is like sharing your computer

![](img/main-slides_full91.png)

![](img/main-slides_full92.png)

_[https://docs\.docker\.com](https://docs.docker.com)_

__Make your environment portable with containers__

![](img/main-slides_full93.png)

__Binder uses the environment documentation file from your Github repo to automatically build a shareable Docker image__

__R__  __uns in the cloud__

__E__  __xecutable__

![](img/main-slides_full94.png)

_[https://tinyurl\.com/jupyter\-binder2\-0](https://tinyurl.com/jupyter-binder2-0)_  _[https://tinyurl\.com/eLife\-binder2\-0](https://tinyurl.com/eLife-binder2-0)_

# Data and code sharing

__What to share?__

Data & code necessary to __validate findings__ & __reproduce results__

Data & code that might be __valuable__ to other researchers/policy makers

Data & code which __cannot be \(easily\) re\-generated__

<span style="color:#FFFFFF"> __Why share?__ </span>

<span style="color:#FFFFFF">Funder or publisher mandates</span>

<span style="color:#FFFFFF">Citation benefits \(Piwowar 2013\,</span>  _[https://doi\.org/10\.7717/peerj\.175](https://doi.org/10.7717/peerj.175)_  <span style="color:#FFFFFF">\)</span>

<span style="color:#FFFFFF">Preserve long\-term access to data</span>

__How to share?__

__Choose open\, persistent\, and non\-proprietary__  __file formats__

__Create and share__  __documentation__  __to enable reuse__

__Include__  __data__  __citations__  __of source data__

__Create rich__  __metadata__

# Data sharing

![](img/main-slides_full95.png)

<span style="color:#2A2A2A">Jonathan D\. Wren;</span>  <span style="color:#2A2A2A"> __URL decay in MEDLINE—a 4\-year follow\-up study__ </span>  <span style="color:#2A2A2A">\, 2008</span>  <span style="color:#2A2A2A"> _Bioinformatics_ </span>

_[https://doi\.org/10\.1093/bioinformatics/btn127](https://doi.org/10.1093/bioinformatics/btn127)_

* __Data Repository__
* Use a data repository not your website\. They provide:
* Persistent identifiers for your data \(like DOI\)
  * Unique & citable
  * Prevents “link rot”
* Persistent access
* Preservation
* Backup
* Management of access
* Versioning
* Licensing

<span style="color:#FFFFFF"> __Data License__ </span>

<span style="color:#FFFFFF">Consider</span>  <span style="color:#FFFFFF"> __Creative Commons licenses__ </span>  <span style="color:#FFFFFF">for data & text</span>

<span style="color:#FFFFFF">Either CC\-0 or CC\-BY</span>

<span style="color:#FFFFFF">Guidance on data licenses available through Digital Curation Center:</span>

_[http://www\.dcc\.ac\.uk/resources/how\-guides\-license\-research\-data](http://www.dcc.ac.uk/resources/how-guides-license-research-data)_

__Code License__

__Consider open source license such as MIT\, BSD\, or Apache license__

__Guidance on software licenses by Karl Broman \(__  _[http://kbroman\.org/steps2rr/pages/licenses\.html](http://kbroman.org/steps2rr/pages/licenses.html)_  __\) and Open Source Initiative \(__  _[https://opensource\.org/licenses](https://opensource.org/licenses)_  __\)__

# Mandated vs. Disciplinary Repositories

__Identify mandated / disciplinary repository:__

__Funder specified repository__

__Institutionally specified data repository__

__Domain or discipline\-specific data repository__

__Find & compare disciplinary repositories through Repository of Research Data Repositories__  _[https://www\.re3data\.org/](http://www.re3data.org/)_

![](img/main-slides_full96.png)

![](img/main-slides_full97.png)

# General Purpose Repositories

__In addition to a specified data repository\, you can make a deposit to a general purpose repository:__

__DataDryad__  _[http://datadryad\.org/](http://datadryad.org/)_  __\(curated digital repository; free to access\, $120 to publish dataset up to 20GB\)__

__Figshare__  _[https://figshare\.com/](https://figshare.com/)_  __\(free digital repository\, 5GB per file limit\)__

__Zenodo__  _[https://zenodo\.org/](https://zenodo.org/)_  __\(free digital repository; 50GB per dataset limit\)__

![](img/main-slides_full98.png)

![](img/main-slides_full99.png)

![](img/main-slides_full100.png)

![](img/main-slides_full101.png)

# Zenodo meets GitHub

![](img/main-slides_full102.png)

![](img/main-slides_full103.png)

# FAIR Data

# Get organized! Be happy!

![](img/main-slides_full104.png)

<span style="color:#333333"> _The Turing Way_ </span>  <span style="color:#333333">project illustration by Scriberia\. Original version on Zenodo\. http://doi\.org/10\.5281/zenodo\.3695300\.</span>

# Data Visualization 
& Analysis

# Data presentation is the foundation of our collective scientific knowledge...

![](img/main-slides_full105.png)

# Figures are especially important. They often show data for key findings.

# Effective figures should...

Immediately convey information about __study design__

Illustrate __important findings__

Allow the reader to __critically evaluate__ the data:Show your data\!

<span style="color:#000000">Weissgerber et al\.</span>  _[10\.1074/jbc\.RA117\.000147](http://www.jbc.org/content/early/2017/10/03/jbc.RA117.000147)_

# The usual way and its flaws...

* Issues:
* Reproducible Workflows?
  * Problems can be avoided by using macros or dashboards
  * However\, who uses these?
* Excel renames Genes
  * <span style="color:#0097A7"> _[Ziemann et al\.\, 2016](https://doi.org/10.1186/s13059-016-1044-7)_ </span>
  * 20% of papers in leading genomic journals contain gene list errors
* Default Plots are often Bar Charts and Line Plots

![](img/main-slides_full106.png)

![](img/main-slides_full107.png)

![](img/main-slides_full108.png)

![](img/main-slides_full109.png)

![](img/main-slides_full110.jpg)

<span style="color:#000000">Weissgerber et al\. 2017</span>  _[10\.1074/jbc\.RA117\.000147](http://www.jbc.org/content/early/2017/10/03/jbc.RA117.000147)_

# Show your data

![](img/main-slides_full111.jpg)

# Never trust data based on the summary statistics alone!There really is NO alternative to showing the actual data

![](img/main-slides_full112.png)

<span style="color:#0097A7"> _[Same Stats\, Different Graphs: Generating Datasets with Varied Appearance and Identical Statistics through Simulated Annealing](https://www.autodeskresearch.com/publications/samestats)_ </span>

![](img/main-slides_full113.gif)

<span style="color:#0097A7"> _[Same Stats\, Different Graphs: Generating Datasets with Varied Appearance and Identical Statistics through Simulated Annealing](https://www.autodeskresearch.com/publications/samestats)_ </span>

# Our interpretation depends on what we see

![](img/main-slides_full114.png)

# Avoid bar charts for continuous data

![](img/main-slides_full115.png)

<span style="color:#000000">Weissgerber et al\.</span>  _[https://doi\.org/10\.1371/journal\.pbio\.1002128](https://doi.org/10.1371/journal.pbio.1002128)_

# How to choose the right plot

![](img/main-slides_full116.png)

_[Weissgerber et al\.\, 2017 JBC](http://www.jbc.org/content/292/50/20592.full)_

# Making effective dot plots

![](img/main-slides_full117.png)

![](img/main-slides_full118.png)

# One step further: Interactive plots

_[Interactive Dot Plot](http://statistika.mfub.bg.ac.rs/interactive-dotplot/)_

_[Interactive Line Graph](http://statistika.mfub.bg.ac.rs/interactive-linegraph/)_

![](img/main-slides_full119.png)

# Some intermediate options

![](img/main-slides_full120.png)

![](img/main-slides_full121.png)

![](img/main-slides_full122.png)

![](img/main-slides_full123.png)

_[https://plot\.ly/create/\#/](https://plot.ly/create/#/)_

# Reproducible research practices enable you to:

<span style="color:#000000">Accurately analyze results</span>

<span style="color:#000000">Share results with future researchers</span>

<span style="color:#000000">Organize experiments productively</span>

<span style="color:#000000">Share techniques</span>

<span style="color:#000000">Share reagents with future researchers</span>

<span style="color:#000000">Accelerate science\!</span>

Tools discussed here should provide you with the framework to make you research more reproducible and will save you time and resources in the long term

# Questions?

Feedback: 
{{postSurveyLink}}

